lib/ConsumerImpl.cc

```diff
@@ -1526,4 +1593,108 @@ void ConsumerImpl::cancelTimers() noexcept {
     checkExpiredChunkedTimer_->cancel(ec);
 }
 
+void ConsumerImpl::processPossibleToDLQ(const MessageId& messageId, ProcessDLQCallBack cb) {
+    auto messages = possibleSendToDeadLetterTopicMessages_.find(messageId);
+    if (!messages) {
+        cb(false);
+        return;
+    }
+
+    // Initialize deadLetterProducer_
+    if (!deadLetterProducer_) {
+        std::lock_guard<std::mutex> createLock(createProducerLock_);
+        if (!deadLetterProducer_) {
+            deadLetterProducer_ = std::make_shared<Promise<Result, Producer>>();
+            ProducerConfiguration producerConfiguration;
+            producerConfiguration.setSchema(config_.getSchema());
+            producerConfiguration.setBlockIfQueueFull(false);
+            producerConfiguration.impl_->setInitialSubscriptionName(
+                deadLetterPolicy_.getInitialSubscriptionName());
+            ClientImplPtr client = client_.lock();
+            if (client) {
+                auto self = get_shared_this_ptr();
+                client->createProducerAsync(
+                    deadLetterPolicy_.getDeadLetterTopic(), producerConfiguration,
+                    [self](Result res, Producer producer) {
+                        if (res == ResultOk) {
+                            self->deadLetterProducer_->setValue(producer);
+                        } else {
+                            LOG_ERROR("Dead letter producer create exception with topic: "
+                                      << self->deadLetterPolicy_.getDeadLetterTopic() << " ex: " << res);
+                            self->deadLetterProducer_.reset();
+                        }
+                    });
+            } else {
+                LOG_WARN(getName() << "Client is destroyed and cannot create dead letter producer.");
```

Comment:
In this case, we should skip the code below because `deadLetterProducer_->getFuture()` will never complete.